presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemorySplit.java

```diff
@@ -49,6 +51,7 @@ public MemorySplit(
         this.partNumber = partNumber;
         this.totalPartsPerWorker = totalPartsPerWorker;
         this.addresses = ImmutableList.copyOf(requireNonNull(addresses, "addresses is null"));
```

Comment:
Aside: for the in-memory connector, shouldn't we check that the addresses list has a single element?